webrender/src/resource_cache.rs

```diff
@@ -801,33 +804,28 @@ impl ResourceCache {
         }
 
         self.cached_images.clear_keys(&mut self.texture_cache, |request| request.key.0 == namespace);
-        self.cached_glyphs.clear_keys(&mut self.texture_cache, |request| request.font.font_key.0 == namespace);
+        self.cached_glyphs.clear_fonts(&mut self.texture_cache, |font| font.font_key.0 == namespace);
     }
 }
 
 pub trait Resource {
-    fn texture_cache_item_id(&self) -> Option<TextureCacheItemId>;
+    fn free(&self, texture_cache: &mut TextureCache);
```

Comment:
There is a 1:1 mapping between a texture cache and a resource cache (which contains these items), so I'm not sure there's an issue here. We could definitely consider tidying this up as a follow up if you're still unsure of the API.